File: app/src/main/java/com/a6v/tjreader/activities/BaseActivity.java
package com.a6v.tjreader.activities;

import android.support.v7.app.AppCompatActivity;

import com.a6v.tjreader.app.App;
import com.a6v.tjreader.app.AppComponent;

public class BaseActivity extends AppCompatActivity {
  protected App getApp() {
    return (App) getApplication();
  }

  protected AppComponent getAppComponent() {
    return getApp().getComponent();
  }
}

File: app/src/main/kotlin/com/a6v/tjreader/activities/BaseActivity.kt
package com.a6v.tjreader.activities

import android.os.Bundle
import com.trello.rxlifecycle.components.support.RxAppCompatActivity
import javax.inject.Inject

/**
 * Created by alexey on 08.03.16.
 */
abstract class BaseActivity : RxAppCompatActivity() {
  @Inject lateinit var retainedState: ActivityRetainedState

  abstract fun initComponent()

  override fun onCreate(savedInstanceState: Bundle?) {
    //component must be initialized before fragments were restored (in super)
    initComponent()
    super.onCreate(savedInstanceState)
  }

  override fun onRetainCustomNonConfigurationInstance(): Any {
    return retainedState
  }
}